src/PureCloud.Client/Apis/TokensApi.cs

```diff
@@ -12,57 +11,101 @@ namespace PureCloud.Client.Apis;
 /// <inheritdoc />
 public sealed class TokensApi : ITokensApi
 {
-    private readonly HttpClient _httpClient;
-    private readonly JsonSerializerOptions _options;
+    private readonly IHttpClientFactory _httpClientFactory;
+    private readonly PureCloudJsonSerializerOptions _options;
 
-    public TokensApi(IHttpClientFactory httpClientFactory, IOptions<PureCloudJsonSerializerOptions> options)
+    public TokensApi(
+        IHttpClientFactory httpClientFactory,
+        IOptions<PureCloudJsonSerializerOptions> options)
     {
-        _httpClient = httpClientFactory.CreateClient(PureCloudConstants.PureCloudClientName);
-        _options    = options.Value.JsonSerializerOptions;
+        _httpClientFactory = httpClientFactory;
+        _options = options.Value;
     }
 
     /// <inheritdoc />
     public async Task<TokenInfo> GetTokensMeAsync(bool? preserveIdleTTL = null, CancellationToken cancellationToken = default)
     {
+        var client = _httpClientFactory.CreateClient(PureCloudConstants.PureCloudClientName);
+        
         var parameters = new NameValueCollection();
-
+        
         if (preserveIdleTTL.HasValue)
         {
             parameters.Add("preserveIdleTTL", UriHelper.ParameterToString(preserveIdleTTL.Value));
         }
 
-        var uri = UriHelper.GetUri("api/v2/tokens/me", parameters);
-
-        var response = await _httpClient.GetAsync(uri, cancellationToken);
+        var response = await client.GetAsync(UriHelper.GetUri("api/v2/tokens/me", parameters), cancellationToken);
 
         response.EnsureSuccessStatusCode();
 
         return await response.Content.ReadFromJsonAsync<TokenInfo>(_options, cancellationToken);
     }
 
     /// <inheritdoc />
-    public async Task DeleteTokenAsync(string userId, CancellationToken cancellationToken = default)
+    public async Task<bool> DeleteTokenAsync(string userId, CancellationToken cancellationToken = default)
     {
         ArgumentException.ThrowIfNullOrEmpty(userId);
 
-        var uri = UriHelper.GetUri($"api/v2/tokens/{userId}", null);
+        var client = _httpClientFactory.CreateClient(PureCloudConstants.PureCloudClientName);
 
-        var response = await _httpClient.DeleteAsync(uri, cancellationToken);
+        var response = await client.DeleteAsync($"api/v2/tokens/{Uri.EscapeDataString(userId)}", cancellationToken);
 
         response.EnsureSuccessStatusCode();
+
+        return true;
```

Comment:
Instead of returning true, return response.IsSucceasfulStatusCode() and remove EnaureSuccessfulStatsCode. Apply the same everywhere